src/components/layer-controller/reducer.js

```diff
@@ -78,53 +85,83 @@ export default function reducer(channels, action) {
       return nextChannels;
     }
-    case 'CHANGE_GLOBAL_CHANNELS_SELECTION': {
-      const { selection, publish } = payload;
+    case 'CHANGE_GLOBAL_CHANNELS_PROPERTIES': {
+      const { update, publish } = payload;
       // Update channel selection for new state.
-      const nextChannels = {};
+      const nextChannels = { ...channels };
       // eslint-disable-next-line no-return-assign
-      Object.keys(channels).forEach(channelId => (
-        nextChannels[channelId] = {
-          ...channels[channelId],
-          selection: {
-            ...channels[channelId].selection,
-            ...selection,
-          },
-        }
-      ));
+      Object.keys(channels).forEach((channelId, i) => {
+        Object.entries(update).forEach(([property, value]) => {
+          const newValue = Array.isArray(value) ? value[i] : value;
+          nextChannels[channelId] = {
+            ...nextChannels[channelId],
+            [property]: getNewChannelProperty(
+              nextChannels[channelId],
+              property,
+              newValue,
+            ),
+          };
+        });
+      });
       // See https://github.com/hubmapconsortium/vitessce-image-viewer/issues/176 for why
       // we don't publish on all changes - only on mouseup (this flag is set in LayerConroller).
+      const layerProps = {};
       if (publish) {
-        const updatedValues = Object.values(nextChannels).map(c => c.selection);
-        const layerProps = { selections: updatedValues };
+        Object.keys(update).forEach((property) => {
+          const updatedValues = Object.values(nextChannels).map(c => c[property]);
+          layerProps[layerProperty[property]] = updatedValues;
+        });
         // Publish deck.gl layer props.
         PubSub.publish(LAYER_CHANGE, { layerId, layerProps });
       }
       return nextChannels;
     }
     case 'ADD_CHANNEL': {
-      const { selection } = payload;
+      const { selection, domain } = payload;
       const channel = {
         selection,
+        domain,
         color: [255, 255, 255],
         visibility: true,
-        slider: [0, 20000],
+        slider: domain,
       };
       const channelId = String(Math.random());
       const nextChannels = { ...channels, [channelId]: channel };
       const layerProps = channelsToLayerProps(nextChannels);
       PubSub.publish(LAYER_CHANGE, { layerId, layerProps });
       return nextChannels;
     }
+    // Because the image layers are asynchronous, hurling a bunch of 'ADD_CHANNEL'
+    // events can lead to unexpected behavior.
+    case 'ADD_CHANNELS': {
+      const { selections, domains } = payload;
+      let nextChannels = { ...channels };
+      selections.forEach((selection, i) => {
+        const domain = domains[i];
+        const channel = {
+          selection,
+          domain,
+          color: VIEWER_PALETTE[i],
+          visibility: true,
+          slider: domain,
+        };
+        const channelId = String(Math.random());
```

Comment:
not super important, but it might be worth moving `String(Math.random())` to a function called `genId`. That way, if we ever want to change how IDs are created, we can just edit it there.